RecoLocalTracker/SiStripZeroSuppression/plugins/SiStripBaselineComparator.cc

```diff
@@ -0,0 +1,177 @@
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "CommonTools/Utils/interface/TFileDirectory.h"
+
+
+//ROOT inclusion
+#include "TH1F.h"
+#include "TProfile.h"
+
+//
+// class decleration
+//
+
+class SiStripBaselineComparator : public edm::one::EDAnalyzer<edm::one::SharedResources> {
+  public:
+    explicit SiStripBaselineComparator(const edm::ParameterSet&);
+    ~SiStripBaselineComparator() override;
+    static void fillDescriptions(edm::ConfigurationDescriptions & descriptions);
+
+
+  private:
+    void beginJob() override ;
+    void analyze(const edm::Event&, const edm::EventSetup&) override;
+    void endJob() override ;
+     
+    edm::EDGetTokenT<edmNew::DetSetVector<SiStripCluster> > srcClusters_;
+    edm::EDGetTokenT<edmNew::DetSetVector<SiStripCluster> > srcClusters2_;
+
+ 
+    edm::Service<TFileService> fs_;
+
+
+    TH1F* h1_nOldClusters_;
+    TH1F* h1_nMatchedClusters_;
+    TH1F* h1_nSplitClusters_;
+    TProfile* h1_matchingMult_;
+    TProfile* h1_matchedWidth_;
+    TProfile* h1_matchedCharges_;
+   
+    const uint16_t nbins_clusterSize_ = 128;
+    const uint16_t min_clusterSize_ = 0;
+    const uint16_t max_clusterSize_ = 128; 
+
+};
+
+
+SiStripBaselineComparator::SiStripBaselineComparator(const edm::ParameterSet& conf){
+ 
+  usesResource(TFileService::kSharedResource);
+  
+  srcClusters_  =  consumes<edmNew::DetSetVector<SiStripCluster> >(conf.getParameter<edm::InputTag>("srcClusters"));
+  srcClusters2_ =  consumes<edmNew::DetSetVector<SiStripCluster> >(conf.getParameter<edm::InputTag>("srcClusters2"));
+ 
+  h1_nOldClusters_ = fs_->make<TH1F>("nOldClusters","nOldClusters;ClusterSize", nbins_clusterSize_, min_clusterSize_, max_clusterSize_);
+  h1_nMatchedClusters_ = fs_->make<TH1F>("nMatchedClusters","nMatchedClusters;ClusterSize", nbins_clusterSize_, min_clusterSize_, max_clusterSize_);
+  h1_nSplitClusters_ = fs_->make<TH1F>("nSplitClusters","nMatchedClusters;ClusterSize; n Split Clusters", nbins_clusterSize_, min_clusterSize_, max_clusterSize_);
+  h1_matchingMult_ = fs_->make<TProfile>("matchingMult","matchingMult;ClusterSize; average number of clusters if split", nbins_clusterSize_, min_clusterSize_, max_clusterSize_);
+  h1_matchedWidth_ = fs_->make<TProfile>("matchedWidth","matchedWidth;ClusterSize;average #Delta Width", nbins_clusterSize_, min_clusterSize_, max_clusterSize_);
+  h1_matchedCharges_ = fs_->make<TProfile>("matchedCharges","matchedCharges;ClusterSize;cluster Charge ratio", nbins_clusterSize_, min_clusterSize_, max_clusterSize_);
+}
+
+
+SiStripBaselineComparator::~SiStripBaselineComparator()
+{
+}
+
+void
+SiStripBaselineComparator::fillDescriptions(edm::ConfigurationDescriptions & descriptions) {
+
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("srcClusters",  edm::InputTag("siStripClusters", "siStripClusters"));
+  desc.add<edm::InputTag>("srcClusters2", edm::InputTag("moddedsiStripClusters", "siStripClusters"));
```

Comment:
[re-posting because github marked my last comment to this as outdated]

this doesn't look right, I think that the siStripClusters branch has no instance defined.
So, it should just be ```edm::InputTag("siStripClusters")```.
I guess ```srcClusters2``` is a placeholder, because afaik ```moddedsiStripClusters``` does not exist in the release.